beacon_node/http_api/tests/tests.rs

```diff
@@ -5318,6 +5354,22 @@ impl ApiTester {
             &[expected_block, expected_head, expected_finalized]
         );
 
+        // Test a block gossip event
+        self.client
+            .post_beacon_blocks(&self.block_gossip)
+            .await
+            .unwrap();
+
+        let block_gossip_events =
+            poll_events(&mut events_future, 1, Duration::from_millis(10000)).await;
+        assert_eq!(
+            block_gossip_events.as_slice(),
+            &[EventKind::BlockGossip(Box::new(BlockGossip {
+                slot: next_slot,
+                block: block_root,
+            }))]
+        );
```

Comment:
I think instead of polling again, you can put the `EventKind::BlockGossip` as the first entry of the list above. In the assert on line 5352 (the one that is currently failing). The test is failing at the moment because the `BlockGossip` event is being emitted for the first block (`self.next_block`), but the test is not expecting it.

If we take this approach, then we don't need `self.block_gossip`, because the gossip event is tested as part of testing `self.next_block`.

Reply:
Fix with your help, thanks Michael!